pkg/install/users.go

```diff
@@ -97,72 +98,21 @@ func EnsureUser(name string, homeDir string) error {
 }
 
 // CreateUser creates a system user with either `adduser` or `useradd` command
-func CreateUser(userName string, homeDir string, shell string) error {
-	var userCmd string
-	var userCmdArgs []string
-
-	logrus.Infof("creating user: %s", userName)
-	_, err := file.GetExecPath("useradd")
-	if err == nil {
-		userCmd = "useradd"
-		userCmdArgs = []string{`--home`, homeDir, `--shell`, shell, `--system`, `--no-create-home`, userName}
-	} else {
-		userCmd = "adduser"
-		userCmdArgs = []string{`--disabled-password`, `--gecos`, `""`, `--home`, homeDir, `--shell`, shell, `--system`, `--no-create-home`, userName}
-	}
-
-	cmd := exec.Command(userCmd, userCmdArgs...)
-	if err := execCmd(cmd); err != nil {
-		return err
+func createUser(userName string, homeDir string, shell string) error {
+	_, err := exec.Command("useradd", `--home`, homeDir, `--shell`, shell, `--system`, `--no-create-home`, userName).Output()
+	if err != nil {
+		_, err = exec.Command("adduser", `--disabled-password`, `--gecos`, `""`, `--home`, homeDir, `--shell`, shell, `--system`, `--no-create-home`, userName).Output()
 	}
-	return nil
+	return err
 }
 
 // DeleteUser deletes system users with either `deluser` or `userdel` command
-func DeleteUser(userName string) error {
-	var userCmd string
-	var userCmdArgs []string
-
-	logrus.Debugf("deleting user: %s", userName)
-	_, err := file.GetExecPath("userdel")
-	if err == nil {
-		userCmd = "userdel"
-		userCmdArgs = []string{userName}
-	} else {
-		userCmd = "deluser"
-		userCmdArgs = []string{userName}
-	}
-
-	cmd := exec.Command(userCmd, userCmdArgs...)
-	if err := execCmd(cmd); err != nil {
-		return err
-	}
-	return nil
-}
-
-// cmd wrapper
-func execCmd(cmd *exec.Cmd) error {
-	logrus.Debugf("executing command: %v", quoteCmd(cmd))
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("failed to run command %s: %v", quoteCmd(cmd), err)
-	}
-	return nil
-}
-
-// parse a cmd struct to string
-func quoteCmd(cmd *exec.Cmd) string {
-	if len(cmd.Args) == 0 {
-		return fmt.Sprintf("%q", cmd.Path)
-	}
-
-	var q []string
-	for _, s := range cmd.Args {
-		q = append(q, fmt.Sprintf("%q", s))
+func deleteUser(userName string) error {
+	_, err := exec.Command("userdel", userName).Output()
+	if err != nil {
```

Comment:
Should this and adduser maybe do something like `if err != exec. ErrNotFound { return err }` to avoid running the other command if the failure in the first one was something else than NotFound?

Reply:
That would be great, but I don't know how to catch that error. The error does no match `os.IsNotExist(err)` nor `exec.ErrNotFound`. The error is `executable file not found in $PATH`, but I'm not sure we want compare against a text string.

Reply:
Figured it out. `exec.ErrNotFound` is what we want.